xonsh/aliases.py

```diff
@@ -22,7 +22,7 @@
 from xonsh.tools import (XonshError, argvquote, escape_windows_cmd_string,
                          to_bool)
 from xonsh.xontribs import xontribs_main
-from xonsh.xoreutils import _which
+from xonsh.xoreutils.which import which_main
```

Comment:
This should be `import xonsh.xoreutils.which` so that this can be a lazy import.  You should only use `from... import...` if the code is in  the same package.